core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -117,7 +117,7 @@ public String toString() {
       this.offsets = ImmutableList.copyOf(offsetList);
       this.parentScanTask = parentScanTask;
       this.targetSplitSize = targetSplitSize;
-      this.splitSizes = new ArrayList<>(offsets.size());
+      this.splitSizes = Lists.newArrayListWithCapacity(offsets.size());
```

Comment:
It looks like most cases in this PR convert to `WithCapacity`, but we typically only use that when we know that the number of elements won't exceed the given size. Otherwise, we use `WithExpectedSize` to allocate some additional space. I think you have that guarantee/expectation here, but if you're converting to one or the other, I'd convert to `WithExpectedSize`.

Reply:
Thank you for your comments. I think what you said is reasonable. When I am modifying the relevant code, I also hesitate to use which method to replace. Later, I saw a description on this method:

> Note: This method will soon be deprecated. Even in the rare case that you do want some amount of padding, it's best if you choose your desired amount explicitly.

This prompted me to use withcapacity,  If use withexpectedsize a lot, will it cause unnecessary waste of resources?
thx.

![image](https://user-images.githubusercontent.com/59213263/145665870-57173659-2cc4-46d2-9566-2a51031f8587.png)

Reply:
The recommendation is to "choose your desired amount explicitly". If you know that there are going to be less than or equal to the requested number of elements, then `WithCapacity` is fine. Otherwise, you'd need to add some additional space. Rather than coming up with a heuristic for the additional space, let's just use `WithExpectedSize`.

Reply:
Done. PTAL, thx.

Reply:
I have fixed it, please help to check the code, thank you. @rdblue 

The main reason is that splitSizes(BaseFileScanTask.OffsetsAwareTargetSplitSizeScanTaskIterator#splitSizes) is of a fixed length, as you said, in this case we should use WithCapacity.